src/sentry/runner/commands/cleanup.py

```diff
@@ -352,7 +352,7 @@ def is_filtered(model: type[Model]) -> bool:
                 else:
                     to_delete_by_project.append(model_tp_tup)
 
-        if project_deletion_query and to_delete_by_project:
+        if project_deletion_query is not None and len(to_delete_by_project):
```

Comment:
Nit: Is the `len` necessary?

Reply:
No, the `len()` isn't necessary. If this resolves the OOM, I'll remove this along with some of the more verbose logging that was added.